Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyAssetCtrlBus.h

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+namespace AzToolsFramework
+{
+    class PropertyAssetCtrlRequests : public AZ::EBusTraits
+    {
+    public:
+        static const AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::Multiple;
+
+        //! Inform any interested components that a new asset has appeared that they may be waiting for.
+        virtual void OnExpectedCatalogAssetAdded(const AZ::Data::AssetId& assetId, const AZ::EntityId& entityId, const AZ::ComponentId& componentId) = 0;
+    };
```

Comment:
This should become a notification bus that lives with the asset editor with a UUID for a bus ID that anyone can connect to. No need to pass in the entity ID or component ID.

Reply:
I've made that change now. I'm now generating a UUID inside the InputConfigurationComponent which is passed across during reflection, avoiding the issue with the inspector being rebuilt every time there's a change. I'm checking through other places that use the asset editor - most have the button disabled until something is loaded.